apps/web/src/components/nav/OrganizationSelect/OrganizationSelect.tsx

```diff
@@ -17,7 +18,8 @@ export function OrganizationSelect() {
   const [isLoading, setIsLoading] = useState<boolean>(false);
   const { addItem, removeItems } = useSpotlightContext();
 
-  const { currentOrganization, organizations, switchOrganization } = useAuth();
+  const { currentOrganization, switchOrganization } = useAuth();
+  const { data: organizations, isLoading: isOrganizationsLoading } = useOrganizations();
```

Comment:
Switching across orgs doesn't depend on the rest of the organizations. You just provide an id. Fetching all organizations is only required in the organization select to see all the other available options.

Does this make the picture a bit clearer?

Reply:
Yes, it does make sense, buy I would like to re visit this later. I think it can be confusing for anyone new coming into the codebase without historical context.